Python/optimizer_bytecodes.c

```diff
@@ -762,6 +764,25 @@ dummy_func(void) {
         ctx->done = true;
     }
 
+    op(_BUILD_TUPLE, (values[oparg] -- tup)) {
+        tup = sym_new_tuple(ctx, oparg);
+        assert(tup != NULL);
+        assert(!ctx->out_of_space);
+        for (int i = 0; i < oparg; i++) {
+            tup->tuple_val[i] = values[i];
+        }
+    }
+
+    op(_BINARY_SUBSCR_TUPLE_INT, (tuple, sub -- res)) {
```

Comment:
This optimization applies to all `BINARY_SUBSCR` variants, although it is much more likely to apply here.